### ISSUE TITLE
[BUG] fix alert VirtOperatorDown

### ISSUE THREAD

**Issue by w13915984028** [harvester/harvester#2615]
**Describe the bug**

When enableing `alertmanager`, continuous `VirtOperatorDown` is seeing.

**To Reproduce**
**Without `alertmanager` enabled:**

In below link, VirtOperatorDown alert is there.
https://the-cluster-vip/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-prometheus:9090/proxy/

**With `alertmanager` enabled:**

1.  Install harvester
2.  Enable alertmanager, add receiver
3.  `VirtOperatorDown` alert comes up

```
192.168.122.195:65393 - - [28/Jul/2022 08:35:32] "HTTP/1.1 POST /" - 200 OK
DATA RECEIVED:2022-07-28 08:45:07.627901
b'{"receiver":"test_py_receiver","status":"firing","alerts":[{"status":"firing","labels":
{"alertname":"VirtOperatorDown","kubernetes_operator_component":"kubevirt","kubernetes_operator_part_of":
"kubevirt","prometheus":"cattle-monitoring-system/rancher-monitoring-prometheus","severity":"critical"},"annotations":{"runbook_url":"https://kubevirt.io/monitoring/runbooks/VirtOperatorDown",
"summary":"All virt-operator servers are down."},"startsAt":"2022-07-28T08:44:37.618Z","endsAt":"0001-01-01T00:00:00Z","generatorURL":"http://rancher-monitoring-prometheus.cattle-monitoring-system:9090/graph?g0.expr=kubevirt_virt_operator_up_total+%3D%3D+0\\u0026g0.tab=1","fingerprint":"9463c3a7f5695fef"}],
"groupLabels":{"alertname":"VirtOperatorDown"},"commonLabels":{"alertname":"VirtOperatorDown","kubernetes_operator_component":"kubevirt","kubernetes_operator_part_of":"kubevirt",
"prometheus":"cattle-monitoring-system/rancher-monitoring-prometheus","severity":"critical"},"commonAnnotations":{"runbook_url":"https://kubevirt.io/monitoring/runbooks/VirtOperatorDown",
"summary":"All virt-operator servers are down."},"externalURL":"http://rancher-monitoring-alertmanager.cattle-monitoring-system:9093","version":"4","groupKey":"{}:{alertname=\\"VirtOperatorDown\\"}","truncatedAlerts":0}\n'

```

But the kube-virt related PODs are working well

```
arvester-system            virt-api-79bc89595-fkhjg                                 1/1     Running     33         12d
harvester-system            virt-api-79bc89595-tgn42                                 1/1     Running     32         12d
harvester-system            virt-controller-6c76d8659-7qgml                          1/1     Running     32         12d
harvester-system            virt-controller-6c76d8659-fsh5l                          1/1     Running     31         12d
harvester-system            virt-handler-25t2t                                       1/1     Running     29         12d
harvester-system            virt-operator-78b84587c9-5dvpd                           1/1     Running     34         12d
```

**Expected behavior**

There should not be such kind of alert.

**Support bundle**

**Environment**
 - Harvester ISO version:  V1.0 and master-head
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): KVM-based

**Additional context**

The issues was caused by that, the `virt-operator-*` POD does not set label `prometheus.kubevirt.io` as `true` , `prometheus` won't fetch POD status which has no such label.

```

harv1:~ # kk get pods -n harvester-system virt-handler-25t2t -o JSON | grep prom
            "prometheus.kubevirt.io": "true"

harv1:~ # kk get pods -n harvester-system virt-controller-6c76d8659-7qgml -o JSON | grep prom
            "prometheus.kubevirt.io": "true"

harv1:~ # kk get pods -n harvester-system virt-operator-78b84587c9-5dvpd -o JSON | grep prom
            "prometheus.kubevirt.io": ""

harv1:~ # kk get pods -n harvester-system harvester-756cd66d6d-dcv2g -o JSON | grep prom
harv1:~ # 
```

The alert `VirtOperatorDown` is defined as such:

```
    - expr: sum(up{namespace='harvester-system', pod=~'virt-operator-.*'}) or vector(0)
      record: kubevirt_virt_operator_up_total
      
    - alert: VirtOperatorDown
      annotations:
        runbook_url: https://kubevirt.io/monitoring/runbooks/VirtOperatorDown
        summary: All virt-operator servers are down.
      expr: kubevirt_virt_operator_up_total == 0
      for: 10m
      labels:
        kubernetes_operator_component: kubevirt
        kubernetes_operator_part_of: kubevirt
        severity: critical
```

More details about Harvester alert:

https://github.com/w13915984028/harvester/blob/fix2517/enhancements/20220720-alertmanager.md

**Comment by w13915984028** [comment]
manually query `virt-controller`, has data
![image](https://user-images.githubusercontent.com/31133476/182392684-dfa85090-4ac7-4a8e-a0ec-48c0b6ac6f4b.png)

manually query `virt-handler`, has data
![image](https://user-images.githubusercontent.com/31133476/182392739-fd97e827-7af3-484f-90f9-1aa05146b33b.png)

manually query `virt-operator`, no data
the grafana hint also shows, there is no pod like `virt-operator-*` is the list
![image](https://user-images.githubusercontent.com/31133476/182392829-81f1cd64-e8d1-468f-877d-96277e6c4537.png)

**Comment by w13915984028** [comment]
manually add label
```
harv1:~ # kk edit pods -n harvester-system virt-operator-78b84587c9-5dvpd
pod/virt-operator-78b84587c9-5dvpd edited

harv1:~ # kk get pods -n harvester-system virt-operator-78b84587c9-5dvpd -o JSON | grep prom
            "prometheus.kubevirt.io": "true"
```

`prometheus` has data now.
![image](https://user-images.githubusercontent.com/31133476/182393308-ed134953-ccf9-42ad-8f43-0a3fe9935fb0.png)

![image](https://user-images.githubusercontent.com/31133476/182393341-a6a603a3-8894-4166-9ef2-24e6c917256e.png)

**Comment by w13915984028** [comment]
the quoted `"true"` in YAML will be converted to string "true", not boolean.

To fix this issue, we need to update both:

`deploy/charts/harvester/dependency_charts/kubevirt-operator/templates/deployment.yaml`
and
`deploy/charts/harvester/charts/kubevirt-operator-0.1.0.tgz`

Indeed, only update to `deploy/charts/harvester/charts/kubevirt-operator-0.1.0.tgz` will be effective.

**Comment by w13915984028** [comment]
Final commi is in PR https://github.com/harvester/harvester/pull/2678

**Comment by w13915984028** [comment]
About the test:

After https://github.com/harvester/harvester/pull/2678 is merged, the `VirtOperatorDown` alert is not existing.

When `alertmanager` is not enabled, existing alerts can still be viewed from :
https://the-cluster-vip/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-prometheus:9090/proxy/

When `alertmanager` is enabled (issue:  [FEATURE] enable/disable alertmanager on demand
https://github.com/harvester/harvester/issues/2518 ,  PR:https://github.com/harvester/harvester-installer/pull/322  ), 

existing alerts can also be viewed from :
https://the-cluster-vip/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-alertmanager:9093/proxy/#/alerts


Comparing, in the Harvester version before the PR, 
In below link, `VirtOperatorDown` alert is there.
https://the-cluster-vip/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-prometheus:9090/proxy/

**Comment by TachunLin** [comment]
Verified **fixed** on `master-4ea12f4f-head` (08/31). Close this issue.

### Result
Since the https://github.com/harvester/harvester/issues/2518 enable/disable alertmanager on demand not yet ready.
We check the alert in the in the prometheus dashboard using above mentioned URL 

* On the latest master with merged this change, we won't see the `VirtOperatorDown` Firing alert 
https://the-cluster-vip/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-prometheus:9090/proxy/
![image](https://user-images.githubusercontent.com/29251855/188106033-6c67c67d-c49d-438d-a492-5b5643746bb4.png)

* Comparing to the previous stable version (v1.0.3), we could see such kind of alert 
![image](https://user-images.githubusercontent.com/29251855/188124934-cc7bba31-ed61-4dfe-ad3d-db228ff0516c.png)

### Test Information
* Test Environment: 1 nodes harvester on local kvm machine
* Harvester version: `master-4ea12f4f-head` (08/31)

### Verify Steps
1. Prepare a single node Harvester with latest master release 
2. Access the https://the-cluster-vip/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-prometheus:9090/proxy/
3. Click the alert tab
4. Clear the Inactive and pending checkbox, only leave the `Firing` 
5. Put this page in idle for at least 30 minutes
6. Check the alert page
7. Prepare a single node Harvester with v1.0.3 release
8. Repeat steps 2 ~ 6 for comparison.
